src/nrtk_explorer/app/core.py

```diff
@@ -213,147 +203,17 @@ def reload_images(self):
-                                        html.Span("Transform Settings", classes="text-h6")
-
-                                    with transforms_content_slot:
-                                        self._transforms_app.settings_widget()
-
-                                    with transforms_actions_slot:
-                                        self._transforms_app.apply_ui()
-
-                            with html.Template(v_slot_after=True):
-                                with quasar.QSplitter(
-                                    v_model=("vertical_split",),
-                                    limits=("[0,100]",),
-                                    horizontal=True,
-                                    classes="inherit-height zero-height",
-                                    before_class="q-pa-md",
-                                    after_class="q-pa-md",
-                                ):
-                                    with html.Template(v_slot_before=True):
-                                        self._embeddings_app.visualization_widget()
-
-                                    with html.Template(v_slot_after=True):
-                                        with html.Div(classes="row q-col-gutter-md"):
-                                            with html.Div(classes="col-6"):
-                                                with quasar.QCard(flat=True, bordered=True):
-                                                    with quasar.QCardSection():
-                                                        html.Span(
-                                                            "Original Dataset", classes="text-h5"
-                                                        )
-
-                                                    with quasar.QCardSection():
-                                                        self._transforms_app.original_dataset_widget()
-
-                                            with html.Div(classes="col-6"):
-                                                with quasar.QCard(
-                                                    flat=True,
-                                                    bordered=True,
-                                                    style="background-color: #ffcdd2;",
-                                                ):
-                                                    with quasar.QCardSection():
-                                                        html.Span(
-                                                            "Transformed Dataset",
-                                                            classes="text-h5",
-                                                        )
-
-                                                    with quasar.QCardSection():
-                                                        self._transforms_app.transformed_dataset_widget()
-
-            self._ui = layout
-
-        return self._ui
```

Comment:
Normally, in Trame and thus in the rest of the source code of this project we aim to keep the UI source code "close" (within) the component that it interacts with, it is a pattern that we followed in the other components.

I wonder if there might be a way for implementing `hot-reloading` without splitting the UI source code from the rest of its component. This is to keep these bits in core.py.

Reply:
I don't agree, the core should use bigger components than sections, spliters, sliders... It may not be just one component but core should focus on its purpose not the little UI details.

Reply:
For example:
- DatasetSelector()
- Embedding()
- Perturbation()

Reply:
I see, that makes sense. I think that moving the UI related to sections splitters and so on makes sense, however, I would keep the UI component bits in core.py (such as the dataset loading widget), because that widget does interacts with callbacks/variables of core. What do you think?

Reply:
The gui object can be outside, but expect inputs that should be passed along the constructor in core.

Reply:
I pushed another commit chunking up the big `build_layout` fucntion.